File: chat-uikit/src/main/java/io/agora/chat/uikit/chat/interfaces/OnRecallMessageResultListener.java
package io.agora.chat.uikit.chat.interfaces;


import io.agora.chat.ChatMessage;

public interface OnRecallMessageResultListener {
    /**
     * Recall successful
     * @param originalMessage The message was unsent
     * @param notification  The notification message
     */
    void recallSuccess(ChatMessage originalMessage, ChatMessage notification);

    /**
     * Recall failed
     * @param code
     * @param errorMsg
     */
    void recallFail(int code, String errorMsg);
}

File: ease-im-kit/src/main/kotlin/com/hyphenate/easeui/feature/chat/interfaces/OnRecallMessageResultListener.kt
package com.hyphenate.easeui.feature.chat.interfaces

import com.hyphenate.easeui.common.ChatMessage


interface OnRecallMessageResultListener {
    /**
     * Recall successful
     * @param originalMessage The message was unsent
     * @param notification  The notification message
     */
    fun recallSuccess(originalMessage: ChatMessage?, notification: ChatMessage?)

    /**
     * Recall failed
     * @param code
     * @param errorMsg
     */
    fun recallFail(code: Int, errorMsg: String?)
}